src/librustdoc/html/static/js/main.js

```diff
@@ -246,14 +257,17 @@ function hideThemeButtonState() {
                 });
             return params;
         },
+        // This function will be replaced once the search has been loaded.
+        getFilterCrates: function() {
```

Comment:
Let's not do this. Instead, let's move `putBackSearch` into `search.js`.

Since `putBackSearch` is referenced from the event listener created in `setup`, we'll also have to move the registration of that event listener into `search.js`.

Reply:
:+1:

Reply:
Actually it's not possible because we load the `search.js` file thanks to this event, making its functions unavailable until they're loaded (so after this event is triggered). I'll try to find another way.

Reply:
In main.js we have:

```
        setup: function() {
            ...
            search_input.addEventListener("focus", function() {
                searchState.putBackSearch(this);
                search_input.origPlaceholder = searchState.input.placeholder;
                search_input.placeholder = "Type your search here.";
                loadSearch();
            });
            search_input.addEventListener("blur", function() {
                search_input.placeholder = searchState.input.origPlaceholder;
            });
```

Those `addEventListener` calls should be moved into `search.js`, in `registerSearchEvents`. I think that should work. At the time that function runs, `putBackSearch` will be available, even if it's defined inside `search.js`.

Reply:
I still need to keep `focus` in here because it loads the script. And to make it smoother, I'll keep the placeholder update before the search scripts are loaded.

Reply:
By the way, I've been thinking: maybe we can do the placeholder with CSS: https://developer.mozilla.org/en-US/docs/Web/CSS/::placeholder

```
#search-input::placeholder {
  content: "Click or press 'S' for search...";
}

#search-input:focus::placeholder {
  content: "Type your search here.";
}
```

Reply:
I thought you didn't want to put text in the CSS? I remember something like that. Well, if so, I'll send a small PR for that. :)

Reply:
Just tried it, you can't set `content` for `::placeholder` unfortunately...